src/tests/Microsoft.PowerFx.Interpreter.Tests/CustomFunctions.cs

```diff
@@ -247,6 +250,119 @@ public static RecordValue Execute()
+        [InlineData("({x: 2, y: 5, a: 8})", false)]
+        [InlineData("({a: 8})", false)]
+        [InlineData("({x: {x : 2}})", false)]
+        public void CustomFunctionRecordTest(string inputRecord, bool isSuccess)
+        {
+            var record = RecordType.Empty().Add(new NamedFormulaType("x", FormulaType.Number)).Add(new NamedFormulaType("y", FormulaType.Number));
+            CustomFunctionRecord(new TestRecordsCustomFunction(record), inputRecord, isSuccess);
+        }
+
+        [Theory]
+        [InlineData("({x: 2})", true)]
+        [InlineData("({x: 2, y: 5})", false)]
+        public void CustomFunctionNonRecordTest(string inputRecord, bool isSuccess)
+        {
+            CustomFunctionRecord(new TestNonRecordCustomFunction(), inputRecord, isSuccess);
+        }
+
+        private void CustomFunctionRecord(ReflectionFunction reflectionFunction, string inputRecord,  bool isSuccess)
+        {
+            var config = new PowerFxConfig();
+            config.AddFunction(reflectionFunction);
+            
+            var engine = new RecalcEngine(config);
+            var func = engine.GetFunctionsByName(reflectionFunction.GetFunctionName());
+            Assert.NotNull(func);
+
+            var check = engine.Check(reflectionFunction.GetFunctionName() + inputRecord);
+
+            FormulaValue result = null; 
+            ConfiguredTaskAwaitable<FormulaValue> resultAsync;
+            string errorMsg = string.Empty;
+            var rc = new RuntimeConfig();
+
+            try
+            {
+                result = check.GetEvaluator().Eval(rc);
+                resultAsync = check.GetEvaluator().EvalAsync(CancellationToken.None, rc).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                errorMsg = ex.ToString();
+            }
+
+            if (isSuccess)
+            {
+                Assert.True(check.IsSuccess);                
+                Assert.IsNotType<ErrorValue>(result);
+                Assert.IsNotType<ErrorValue>(resultAsync);
+            }
+            else
+            {
+                Assert.False(check.IsSuccess);
+                Assert.Contains("invalid arguments", errorMsg);
+            }
+        }
+
+        // Must have "Function" suffix. 
+        private class TestRecordsCustomFunction : ReflectionFunction
+        {
+            public TestRecordsCustomFunction(RecordType recordType)
+                : base("RecordsTest", recordType, recordType)
+            {
+            }
+
+            // Must have "Execute" method. 
+            public static RecordValue Execute(RecordValue recordValue)
+            {
+                return FormulaValue.NewRecordFromFields(recordValue.Type, recordValue.Fields);
+            }
+        }
+
+        // Must have "Function" suffix. 
+        private class TestNonRecordCustomFunction : ReflectionFunction
+        {
+            public TestNonRecordCustomFunction()
```

Comment:
>TestNonRecordCustomFunction [](http://example.com/codeflow?start=19&length=27)

This change is checking the input types of functions; this function has an input type of type record, just like the previous one. Do we need two of them?

Reply:
remove this.